test/test_fork.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2022-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test that pymongo is fork safe."""
+
+import os
+from multiprocessing import Pipe
+from test import IntegrationTest, client_context
+from unittest import skipIf
+
+from bson.objectid import ObjectId
+from pymongo.lock import _insertion_lock
+
+
+@client_context.require_connection
+def setUpModule():
+    pass
+
+
+# Not available for versions of Python without "register_at_fork"
+@skipIf(
+    not hasattr(os, "register_at_fork"), "register_at_fork not available in this version of Python"
+)
+class TestFork(IntegrationTest):
+    def test_lock_client(self):
+        """
+        Forks the client with some items locked.
+        Parent => All locks should be as before the fork.
+        Child => All locks should be reset.
+        """
+
+        def exit_cond():
+            with _insertion_lock:
+                return 0  # success
```

Comment:
It might be better to run an operation on the client in the child process here, like:
```
self.client.admin.command('ping')
```